### PR TITLE
export submissions to GFF3 format

### PR DESCRIPTION
**Pull request by hargup**
- [x] Change indentation to 2 spaces.
- [x] Verify the generated GFF3 file using http://genometools.org/cgi-bin/gff3validator.cgi
- [x] Handle more than one submission
- [x] Handle more than one transcript in a single submission
- [x] Create a black list of non standard subfeature type to ignore while exporting them.
- [x] Fix the phase calculation
- [x] Query `Task` and `Submission` to find out all the submission corresponding to a `GFF` given by the user from the database.
- [x] Re-factor code to reduce redundancy.
- [x] Calculate the `end` part for the `##sequence-region` meta-data
- [x] Rebase and clean up the history.
- [x] Generate the names of the features on the client side.

### CODE REVIEW

**Comment by yeban** [comment]
@hargup Generating names client side is now in master. Please could you rebase the PR on master and quickly verify if everything is good to merge?

File: services/exporter.rb [hunk 1]
```diff
@@ -0,0 +1,137 @@
...
+
+      sr_data[seq_id_index[seq_id]] << transcript_data
+    end
+
+    sr_str = []
+    sr_data.each_with_index do |file, i|
+      sr_str << []
+      seq_id = seq_id_index.invert[i]
+      start = 1
+      stop = (RefSeq.where seq_id: seq_id).first[:length]
```
Comment (yeban) [inline services/exporter.rb]: ping!

File: services/exporter.rb [hunk 2]
```diff
@@ -0,0 +1,154 @@
...
+
+    @subfeature_count = Hash.new(0)
+  end
+
+  attr_reader :genome
+
+  # Returns all the submissions objects for a given GFF3 file.
+  def submissions
+    ref_seqs = @genome.ref_seqs.map(&:seq_id)
+    task_ids  = Task.where(ref_seq_id: ref_seqs).map(&:id)
```
Comment (yeban) [inline services/exporter.rb]: unnecessary whitespace here after `task_ids`

**Comment by yeban** [comment]
@hargup Alternative splicing hasn't yet been taken into account.

**Comment by hargup** [comment]
@yeban have a look now.

**Comment by yeban** [comment]
Thanks a ton!
